tests/integration/test_tls.py

```diff
@@ -250,6 +257,107 @@ async def test_mtls(ops_test: OpsTest):
     assert max_offset == str(num_messages)
 
 
+@pytest.mark.abort_on_fail
+async def test_truststore_live_reload(ops_test: OpsTest):
+    """Tests truststore live reload functionality using kafka-python client."""
+    requirer = "other-req/0"
+    test_msg = {"test": 123456}
+
+    await ops_test.model.deploy(
+        TLS_NAME, channel="stable", series="jammy", application_name="other-ca"
```

Comment:
**thought**: I did not bother making a comment in the first PR, but I changed my mind. `self-signed-certificates` is broken with our charm since rev 164. That's why we pin the revision to rev 163 on L56. Since this deployment tracks stable (rev 155) it's good for now, but you must know that it could break at some point.
No need for change though, I just wanted you to have the context to make it easier for the future you

Reply:
> **thought**: I did not bother making a comment in the first PR, but I changed my mind. `self-signed-certificates` is broken with our charm since rev 164. That's why we pin the revision to rev 163 on L56. Since this deployment tracks stable (rev 155) it's good for now, but you must know that it could break at some point. No need for change though, I just wanted you to have the context to make it easier for the future you

Good point, I didn't know that. After all, I prefer a controlled test env. so I did pin the revisions.